File: app/src/main/java/org/test/kotlin/demo/repos/search/SearchReposModule.java
package org.test.kotlin.demo.repos.search;

import dagger.Binds;
import dagger.Module;

@Module
public interface SearchReposModule {

    @Binds
    SearchReposContract.View bindView(SearchReposFragment impl);

    @Binds
    SearchReposContract.Presenter bindPresenter(SearchReposPresenter impl);

}

File: app/src/main/java/org/test/kotlin/demo/repos/search/SearchReposModule.kt
package org.test.kotlin.demo.repos.search

import dagger.Binds
import dagger.Module

@Module
interface SearchReposModule {

    @Binds
    fun bindView(impl: SearchReposFragment): SearchReposContract.View

    @Binds
    fun bindPresenter(impl: SearchReposPresenter): SearchReposContract.Presenter

}
